source/extensions/filters/http/jwt_authn/filter_config.h

```diff
@@ -97,12 +110,24 @@ class FilterConfig : public Logger::Loggable<Logger::Id::config>, public AuthFac
   const Extractor& getExtractor() const { return *extractor_; }
 
   // Finds the matcher that matched the header
-  virtual const Verifier* findVerifier(const Http::HeaderMap& headers) const {
+  virtual const Verifier* findVerifier(const Http::HeaderMap& headers,
+                                       const envoy::api::v2::core::Metadata& metadata) const {
     for (const auto& pair : rule_pairs_) {
       if (pair.matcher_->matches(headers)) {
         return pair.verifier_.get();
       }
     }
+    if (metadata_path_.size() > 0 && metadata_verifiers_.size() > 0) {
+      const auto& value =
+          Envoy::Config::Metadata::metadataValue(metadata, metadata_filter_, metadata_path_);
+      if (value.kind_case() == ProtobufWkt::Value::kStringValue) {
+        ENVOY_LOG(debug, "use metadata_rules key: {}", value.string_value());
```

Comment:
Maybe be a bit more descriptive, indicating this is a JWT verifier selection.

Reply:
Done